test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs

```diff
@@ -268,8 +268,28 @@ public void It_migrates_and_builds_projects_in_global_json(string path, string p
             outputsIdentical.Should().BeTrue();
             VerifyAllMSBuildOutputsRunnable(projectDirectory);
          }
+        
+        [Theory]
+        // https://github.com/dotnet/cli/issues/4313
+        [InlineData("TestDirWithNoProjects", true)]
+        [InlineData("TestDirWithNoProjects", false)]
+        public void It_migrates_no_projects_found(string projectDir, bool useGlobalJson)
+        {
+            // The migrate command should return 1 for failure.
+            const int ExpectedResult = 1;
```

Comment:
Since you're using MigrateCommand directly (in-proc) through the `MigrateProject` function, you'll have to catch the exception I think, and yes I think you should verify the message on the exception.